File: src/libs/ThirdParty.VBSuppliers/Properties/ExpediaRapid/SerializableClasses/Search/RoomRate.vb
Imports Newtonsoft.Json
Namespace ExpediaRapid.SerializableClasses.Search

    Public Class RoomRate

        <JsonProperty("id")>
        Public Property RateID As String

        <JsonProperty("available_rooms")>
        Public Property AvailableRooms As Integer

        <JsonProperty("refundable")>
        Public Property IsRefundable As Boolean

        <JsonProperty("fenced_deal")>
        Public Property IsFencedDeal As Boolean

        <JsonProperty("fenced_deal_available")>
        Public Property HasFencedDealAvailable As Boolean

        <JsonProperty("deposit_required")>
        Public Property DepositRequired As Boolean

        <JsonProperty("merchant_of_record")>
        Public Property MerchantOfRecord As String

        <JsonProperty("amenities")>
        Public Property Amenities As New Dictionary(Of String, Amenities)

        <JsonProperty("links")>
        Public Property RateLinks As New Dictionary(Of String, Link)

        <JsonProperty("bed_groups")>
        Public Property BedGroupAvailabilities As New Dictionary(Of String, BedGroupAvailability)

        <JsonProperty("cancel_penalties")>
        Public Property CancelPenalities As New List(Of CancelPenalty)

        <JsonProperty("occupancy_pricing")>
        Public Property OccupancyRoomRates As New Dictionary(Of String, OccupancyRoomRate)

        <JsonProperty("promotions")>
        Public Property Promotions As Promotions

    End Class

End Namespace

File: src/libs/iVectorOne.Suppliers/ExpediaRapid/SerializableClasses/Search/RoomRate.cs
namespace iVectorOne.Suppliers.ExpediaRapid.SerializableClasses.Search
{
    using System.Collections.Generic;
    using Newtonsoft.Json;

    public class RoomRate
    {

        [JsonProperty("id")]
        public string RateID { get; set; }

        [JsonProperty("available_rooms")]
        public int AvailableRooms { get; set; }

        [JsonProperty("refundable")]
        public bool IsRefundable { get; set; }

        [JsonProperty("deposit_required")]
        public bool DepositRequired { get; set; }

        [JsonProperty("merchant_of_record")]
        public string MerchantOfRecord { get; set; }

        [JsonProperty("amenities")]
        public Dictionary<string, Amenities> Amenities { get; set; } = new Dictionary<string, Amenities>();

        [JsonProperty("links")]
        public Dictionary<string, Link> RateLinks { get; set; } = new Dictionary<string, Link>();

        [JsonProperty("bed_groups")]
        public Dictionary<string, BedGroupAvailability> BedGroupAvailabilities { get; set; } = new Dictionary<string, BedGroupAvailability>();

        [JsonProperty("cancel_penalties")]
        public List<CancelPenalty> CancelPenalities { get; set; } = new List<CancelPenalty>();

        [JsonProperty("occupancy_pricing")]
        public Dictionary<string, OccupancyRoomRate> OccupancyRoomRates { get; set; } = new Dictionary<string, OccupancyRoomRate>();

        [JsonProperty("promotions")]
        public Promotions Promotions { get; set; }

    }

}